OpenRA.Mods.Common/Activities/Air/ReturnToBase.cs

```diff
@@ -175,62 +118,37 @@ public override Activity Tick(Actor self)
 
 						return this;
 					}
-					else
-					{
-						QueueChild(self,
-							new Fly(self, Target.FromActor(nearestResupplier), WDist.Zero, aircraft.Info.WaitDistanceFromResupplyBase, targetLineColor: Color.Green),
-							true);
-
-						QueueChild(self, new FlyCircle(self, aircraft.Info.NumberOfTicksToVerifyAvailableAirport), true);
-						return this;
-					}
-				}
-				else if (nearestResupplier == null && aircraft.Info.VTOL && aircraft.Info.LandWhenIdle)
-				{
-					// Using Queue instead of QueueChild here is intentional, as we want VTOLs with LandWhenIdle to land and stay there in this situation
-					Cancel(self);
-					if (aircraft.Info.TurnToLand)
-						Queue(self, new Turn(self, aircraft.Info.InitialFacing));
 
-					Queue(self, new Land(self));
-					return NextActivity;
-				}
-				else
-				{
-					// Prevent an infinite loop in case we'd return to the activity that called ReturnToBase in the first place. Go idle instead.
-					Cancel(self);
-					return NextActivity;
+					QueueChild(self, new Fly(self, Target.FromActor(nearestResupplier), WDist.Zero, aircraft.Info.WaitDistanceFromResupplyBase, targetLineColor: Color.Green),
+							true);
+					QueueChild(self, new FlyCircle(self, aircraft.Info.NumberOfTicksToVerifyAvailableAirport), true);
+					return this;
 				}
-			}
-
-			var exit = dest.FirstExitOrDefault(null);
-			var offset = exit != null ? exit.Info.SpawnOffset : WVec.Zero;
 
-			if (aircraft.Info.VTOL || aircraft.Info.CanHover)
-				QueueChild(self, new Fly(self, Target.FromPos(dest.CenterPosition + offset)), true);
-			else
-			{
-				var turnRadius = Fly.CalculateTurnRadius(aircraft.Info.Speed, aircraft.Info.TurnSpeed);
-
-				QueueChild(self, new Fly(self, Target.FromPos(w1), WDist.Zero, new WDist(turnRadius * 3)), true);
-				QueueChild(self, new Fly(self, Target.FromPos(w2)), true);
-
-				// Fix a problem when the airplane is sent to resupply near the airport
-				QueueChild(self, new Fly(self, Target.FromPos(w3), WDist.Zero, new WDist(turnRadius / 2)), true);
+				// Prevent an infinite loop in case we'd return to the activity that called ReturnToBase in the first place. Go idle instead.
+				self.CancelActivity();
+				return NextActivity;
 			}
 
 			if (ShouldLandAtBuilding(self, dest))
 			{
-				aircraft.MakeReservation(dest);
-
-				if (aircraft.Info.VTOL && aircraft.Info.TurnToDock)
-					QueueChild(self, new Turn(self, aircraft.Info.InitialFacing), true);
+				var exit = dest.FirstExitOrDefault(null);
+				var offset = exit != null ? exit.Info.SpawnOffset : WVec.Zero;
+				if (aircraft.Info.TurnToDock)
+					facing = aircraft.Info.InitialFacing;
+				if (!aircraft.Info.VTOL)
+					facing = 196;
```

Comment:
64x3, so 192.

Reply:
This should be exposed to YAML longterm.

Reply:
@GraionDilach That was my plan for a follow up, yes.

Reply:
Fixed the value.